### DISCUSSION TITLE
Installing in a virtual environment

### DISCUSSION THREAD

**Discussion by SMDeBellis** [python-poetry/poetry#9566] [Q&A]
In the Installation portion of the documentation it states:

> Poetry should always be installed in a dedicated virtual environment to isolate it from the rest of your system. It should in no case be installed in the environment of the project that is to be managed by Poetry. This ensures that Poetry’s own dependencies will not be accidentally upgraded or uninstalled. (Each of the following installation methods ensures that Poetry is installed into an isolated environment.) In addition, the isolated virtual environment in which poetry is installed should not be activated for running poetry commands.

I have created a virtual environment using python -m venv my-env, activated the environment, and then ran the curl install command but I can still poetry --version in a different shell.

My question is, was the above documentation referencing a different type of virtual environment? Should I be able to run poetry commands outside of the python venv?

**Comment by radoering** [comment]
If you are using pipx or the curl command (aka official installer), you do not have to create a venv yourself because pipx / the installer do it for you. You only have to create the venv yourself if you use the manual installation method.

> Should I be able to run poetry commands outside of the python venv?

Yes, because you should not activate the environment of Poetry. (Or you should at least deactivate it after installing Poetry.) Please continue reading the documentation page. There are detailed instructions for each of the three methods. You do not have to do additional steps in advance.
